src/js/components/FrameworkConfigurationForm.js

```diff
@@ -0,0 +1,315 @@
+
+    return Object.keys(errorSchemaLevel)
+      .map(key => this.getTotalErrorsForLevel(errorSchemaLevel[key]))
+      .reduce((a, b) => a + b, currentLevelErrors);
+  }
+
+  handleFormChange(form) {
+    const { formData, errorSchema } = form;
+
+    const formErrors = {};
+    Object.keys(errorSchema).forEach(tab => {
+      formErrors[tab] = this.getTotalErrorsForLevel(errorSchema[tab]);
+    });
+
+    this.props.onFormErrorChange(formErrors);
+    this.props.onFormDataChange(formData);
+    this.setState({ errorSchema });
+  }
+
+  validate(formData, errors) {
+    const { packageDetails } = this.props;
+
+    this.writeErrors(formData, packageDetails.config, false, errors);
+
+    return errors;
+  }
+
+  formatErrorForJSON(errorSchema, path) {
+    if (!errorSchema.__errors) {
+      return [];
+    }
+    const currentLevelErrors = errorSchema.__errors.map(message => {
+      return { message, path };
+    });
+
+    return Object.keys(errorSchema)
+      .map(key => this.formatErrorForJSON(errorSchema[key], path.concat(key)))
+      .reduce((a, b) => a.concat(b), currentLevelErrors);
+  }
+
+  getErrorsForJSONEditor() {
+    const { errorSchema } = this.state;
+
+    if (!errorSchema) {
+      return [];
+    }
+
+    return this.formatErrorForJSON(errorSchema, []);
+  }
+
+  render() {
+    const {
+      packageDetails,
+      jsonEditorActive,
+      formData,
+      focusFieldPath,
+      deployErrors
+    } = this.props;
+    const activeTab = focusFieldPath[0];
+    const schema = packageDetails.config;
+    const jsonEditorErrors = this.getErrorsForJSONEditor();
+
+    const TitleField = props => {
+      return <DefaultTitleField {...props} required={false} />;
+    };
+
+    const jsonEditorPlaceholderClasses = classNames(
+      "modal-full-screen-side-panel-placeholder",
+      { "is-visible": jsonEditorActive }
+    );
+    const jsonEditorClasses = classNames("modal-full-screen-side-panel", {
+      "is-visible": jsonEditorActive
+    });
+
+    let errorsAlert = null;
+    if (deployErrors) {
+      errorsAlert = <CosmosErrorMessage error={deployErrors} />;
+    }
+
+    return (
+      <div className="flex flex-item-grow-1">
+        <div className="create-service-modal-form__scrollbar-container modal-body-offset gm-scrollbar-container-flex">
+          <PageHeaderNavigationDropdown
+            handleNavigationItemSelection={this.handleDropdownNavigationSelection.bind(
+              this
+            )}
+            items={this.getDropdownNavigationList()}
+          />
+          <div className="modal-body-padding-surrogate create-service-modal-form-container">
+            <div className="create-service-modal-form container container-wide">
+              <Tabs
+                activeTab={activeTab}
+                handleTabChange={this.handleTabChange.bind(this)}
+                vertical={true}
+                className={"menu-tabbed-container-fixed"}
```

Comment:
@Poltergeist I'm pretty sure that you or @jfurrow fixed this in the service form. Could you please help @ahoskins apply the _same_ fix?

Reply:
As far as I can tell, this has the same styling as the service form. It will ellipsis when a single word is longer than the space available. And it will wrap when there are many words it can split on. *But I've added a new class `padded-right` to the menu-tabbed, which gives it 5px of spacing.*  Now it looks like the below.

Ellipsis:
![screen shot 2017-10-11 at 3 26 15 pm](https://user-images.githubusercontent.com/1527504/31469927-958017c0-ae98-11e7-90f5-d85c077dc83c.png)

Wrap:
![screen shot 2017-10-11 at 3 25 17 pm](https://user-images.githubusercontent.com/1527504/31469882-70d416a6-ae98-11e7-90cc-57e8fcc01685.png)

Reply:
👋 Oh hi! Lookin' good!

I think I only added padding for the container names when configuring a multi-container service. Maybe I'm wrong about that — but in any case I should have applied it to all of these tabs!

Reply:
Hey John Tanks for the Feedback. :-) This what you see there is basically the new advanced configuration view which Andrew is about to finish and I would have suggested the same. But Thanks for jumping in, that was unexpected.

Reply:
@jfurrow are you ready to come back yet?